workflow-templates/trivy.yml

```diff
@@ -0,0 +1,46 @@
+name: trivy
+
+on:
+  pull_request:
+    branches: [ "main", "develop" ]
+  schedule:
+    - cron: '0 0 1 * *'
+  push:
+    branches: [ "main", "develop" ]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    permissions:
+      contents: read 
+      security-events: write
+
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -t docker.io/kbase/trivy:scan .
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: "kbase/trivy:scan"
+          format: "sarif"
+          output: "trivy-results.sarif"
+          timeout: "20m"
+          ignore-unfixed: true
+          severity: 'HIGH,CRITICAL'
+          limit-severities-for-sarif: true
+          exit-code: '1' # Fail Step if CVE of specified severity is detected
```

Comment:
Again, this means that any repo that adds this scan will likely fail the scan for a good while unless it's pretty new

Reply:
Yes, sounds good to me. Users will have this opportunity to fix it the next time they do a build. However, this will not block any build or PR.

Reply:
Well, it will if 
![image](https://github.com/user-attachments/assets/fb238090-7c62-4118-81b4-605f1b63d108)
is set. I don't think we do that though?

Reply:
* I have not set it
* A PR gets a review by a person, and if a person sees that the trivy scan failed, they can block the PR.

Reply:
To be clear I don't think it should block PRs